### PR TITLE
Start implementing Assets microservice

### PR DESCRIPTION
**Pull request by Sheko1**
# This PR adds functionality to receive bank accounts balances and transactions history.

# Endpoints
**/balances** - parameters in headers - **key=nordigen value=requisition_id** - returns balances for every bank account in the requisition.
**/transactions** - parameters in headers - **key=nordigen value=requisition_id** - returns only booked transactions for every bank account in the requisition.

# How to run
Install requrements and run the wsgi.py file

### CODE REVIEW

File: assets/src/blueprints/balances.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            if not accounts:
+                data['nordigen'] = 'No bank accounts'
+
+            else:
+                data['nordigen'] = {}
+
+                for account in accounts:
+                    data['nordigen'][account] = nordigen.get_account_balances(account).get('balances')[0]
+
+        return data
```
Comment (DimoDimchev) [inline assets/src/blueprints/balances.py:26]: Shouldn't this return something like 'No Nordigen key was provided'?

File: assets/src/blueprints/transactions.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            if not accounts:
+                data['nordigen'] = 'No bank accounts'
+
+            else:
+                data['nordigen'] = {}
+
+                for account in accounts:
+                    data['nordigen'][account] = nordigen.get_account_transactions(account).get('transactions').get('booked')
+
+        return data
```
Comment (DimoDimchev) [inline assets/src/blueprints/transactions.py:26]: Same remark as the one above

File: assets/src/blueprints/balances.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        if nordigen_key:
+            accounts = nordigen.list_accounts(nordigen_key).get('accounts')
+            if not accounts:
+                data['nordigen'] = 'No bank accounts'
+
+            else:
+                data['nordigen'] = {}
+
+                for account in accounts:
+                    data['nordigen'][account] = nordigen.get_account_balances(account).get('balances')[0]
```
Comment (DimoDimchev) [inline assets/src/blueprints/balances.py:24]: You should consider moving this logic into a separate function and just calling it from inside the endpoint. This way we can combine Nordigen, Coinbase, Binance and Yodlee functionality into a single endpoint without making the code difficult to understand. The same goes for the transactions logic.

File: assets/src/blueprints/balances.py (hunk 1 above)
Comment (Sheko1) [reply assets/src/blueprints/balances.py:26]: Every linked bank account in nordigen has unique id, its auto generated when user link his account and we can do a request with that id. We can see all user linked bank accounts id's in his requisition so we need only to store requisition id in the database.
accounts = nordigen.list_accounts(nordigen_key).get('accounts') - how i get the accounts id's, nordigen_key is a requisition id.
The only thing we need to use nordigen api is the token that you need to generate in your account.

File: assets/src/blueprints/balances.py (hunk 3 above)
Comment (Sheko1) [reply assets/src/blueprints/balances.py:24]: I will to that thank you for the feedback.

File: assets/src/blueprints/balances.py (hunk 1 above)
Comment (DimoDimchev) [reply assets/src/blueprints/balances.py:26]: I don't think we're on the same page. I see you're making an if-else statement to check if the token is present in the headers. What I'm asking is why doesn't `data` get filled with `Token not in request header` or something like that if the token isn't present

File: assets/src/blueprints/balances.py (hunk 3 above)
Comment (Sheko1) [reply assets/src/blueprints/balances.py]: I fixed the problems you mentioned please review again!

File: assets/src/blueprints/balances.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from utils import nordigen
+
+
+balances = Blueprint('balances', __name__)
+api = Api(balances)
+
+
+class GetBalances(Resource):
+    def get(self):
+        data = {
```
Comment (DimoDimchev) [inline assets/src/blueprints/balances.py:12]:
```suggestion
        data = {}
        if request.headers.get('nordigen'):
           data['nordigen'] =  nordigen.get_account_balances(request.headers.get('nordigen'))
```

File: assets/src/blueprints/balances.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            'nordigen': nordigen.get_account_balances(request.headers.get('nordigen')),
+        }
```
Comment (DimoDimchev) [inline assets/src/blueprints/balances.py:14]:
```suggestion
         data = {}
        if request.headers.get('nordigen'):
           data['nordigen'] =  nordigen.get_account_balances(request.headers.get('nordigen'))
        else:
           data['nordigen'] = 'No key supplied'
```

**Review by DimoDimchev — APPROVED** [review]

**Comment by DimoDimchev** [comment]
Good job 👍

**Review by KoliosterNikolayIliev — APPROVED** [review]

File: assets/utils/nordigen.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    accounts = requisition.get('accounts')
+
+    if not accounts:
+        return {'message': 'No bank accounts'}, False
+
+    return accounts, True
+
+
+def get_account_balances(requisition_id):
+    accounts = get_bank_accounts(requisition_id)
```
Comment (smanilov) [inline assets/utils/nordigen.py:35]: Please add a comment describing the structure of the result.

**Pull request merged**